httpcore/_backends/curio.py

```diff
@@ -0,0 +1,213 @@
+import select
+import socket
+from ssl import SSLContext, SSLSocket
+from typing import Dict, Optional, Type, Union
+
+import curio
+import curio.io
+
+from .._exceptions import (
+    ConnectError,
+    ConnectTimeout,
+    ReadError,
+    ReadTimeout,
+    TimeoutException,
+    WriteError,
+    WriteTimeout,
+    map_exceptions,
+)
+from .._types import TimeoutDict
+from .._utils import get_logger
+from .base import AsyncBackend, AsyncLock, AsyncSemaphore, AsyncSocketStream
+
+logger = get_logger("curio_backend")
+
+ONE_DAY_IN_SECONDS = 60 * 60 * 24
+
+
+async def wrap_ssl_client(
+    sock: curio.io.Socket,
+    ssl_context: SSLContext,
+    server_hostname: bytes,
+) -> curio.io.Socket:
+    kwargs = {
+        "server_hostname": server_hostname,
+        "do_handshake_on_connect": sock._socket.gettimeout() != 0.0,
+    }
+
+    socket = curio.io.Socket(ssl_context.wrap_socket(sock._socket, **kwargs))
+    await socket.do_handshake()
+
+    return socket
+
+
+def convert_timeout(value: Optional[float]) -> int:
+    return int(value) if value is not None else ONE_DAY_IN_SECONDS
```

Comment:
Right? 🤔

```suggestion
def convert_timeout(value: Optional[float]) -> float:
    return float(value) if value is not None else ONE_DAY_IN_SECONDS
```

Reply:
No, as you write above, `curio` do not accept `float` as a timeout, so it's `int`

Reply:
It actually _does_ accept float numbers — the error message in response to using `float("inf")` is misleading.

```python
import curio

curio.run(curio.sleep(1.5))  # Sleeps to 1.5 seconds
```

Reply:
You are right, I checked it again. Changing it

_For me_: first check, then say, not vice versa